andela/ah-magnificent6
# #161199151 status code for using non-existing email address to receive link to reset password

#### What does this PR do?
- changes status code for a non-existing email address from 200 to 404

#### Description of tasks to be completed
- change status code from 200 to 404 for a non-existing email address
- update tests to reflect this change
- read URL of the request object to include it in the link send to a user to reset password

#### How should this be manually tested?
- Clone [this](https://github.com/andela/ah-magnificent6.git) repository
- Copy everything on *.env-sample* to  *.env* [and set the environment variables]
- Create and activate a [virtual environment](https://docs.python.org/3/library/venv.html) inside the project base directory.
- Run the command `git checkout bg-forgot-password-status-code-161199151`
- Install the project dependencies by running `pip install -r requirements.txt`
- Run the server by issuing the command `python manage.py runserver`
- Using ([Postman](https://www.getpostman.com/) or your favorite REST API client, 
- Access  `/api/accounts/forgot_password/` using a `notexisting@notthere.com` email address
- Examine the response you get; it should have a status code of 404 and a message

#### Any background context you want to provide?
Using correct status code for various server response is vital in any application because it helps one know the context behind a given response.

#### What are the relevant pivotal tracker stories?
[Fix status code error](https://www.pivotaltracker.com/story/show/161199151)

#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/9263906/46906503-31b0e080-cf0d-11e8-9645-9bbc6cfd106e.png)
